pages/api/v1/answers.ts

```diff
@@ -386,46 +396,60 @@ async function Angela( // Good ol' Angela
       // Add assistant message to nonSystemMessages
       nonSystemMessages.push(newMessage);
 
+      const toConfirm: { actionId: number; parameters: object }[] = [];
       // Call endpoints here
       for (const command of mostRecentParsedOutput.commands) {
         const chosenAction = actions.find((a) => a.name === command.name);
         if (!chosenAction) {
           throw new Error(`Action ${command.name} not found!`);
         }
-        let out = await httpRequestFromAction(
-          chosenAction,
-          command.args,
-          org,
-          streamInfo,
-          reqData.user_api_key ?? ""
-        );
-        console.log("Output from API call:", out);
-        // Post-processing
-        if (Array.isArray(out)) {
-          out = deduplicateArray(out);
-        }
-        const keys = chosenAction.keys_to_keep;
+
+        const actionToHttpRequest: ActionToHttpRequest = {
+          action: chosenAction,
+          parameters: command.args,
+          organization: org,
+          stream: streamInfo,
+          userApiKey: reqData.user_api_key ?? "",
+        };
+
         if (
-          keys &&
-          Array.isArray(keys) &&
-          keys.every((k) => typeof k === "string")
+          ["get", "head", "options", "connect"].includes(
+            chosenAction.request_method!.toLowerCase()
+          )
         ) {
-          out = filterKeys(out, keys as string[]);
+          let out = await httpRequestFromAction(actionToHttpRequest);
+          out = processAPIoutput(out, chosenAction);
+          console.log("Output from API call:", out);
+          streamInfo({
+            role: "function",
+            name: command.name,
+            content: JSON.stringify(out, null, 2),
+          });
+          const renderableOutput =
+            "\n\n" + command.name + " output:\n" + convertToRenderable(out);
+          nonSystemMessages.push({
+            role: "function",
+            name: command.name,
+            content: renderableOutput,
+          });
+        } else {
```

Comment:
Not 100% where in the code this bug is, but I found a bug - haven't clicked yes or no yet:

Edit: Tested some more and this happens every time

Backend log output:
```
GPT output cost:  0.00522
[TypeError: Failed to parse URL from ]
```
![Screenshot from 2023-07-13 20-19-16](https://github.com/Superflows-AI/superflows/assets/33871096/246b8bdd-97f9-487f-9239-dd97bc1f83a8)

Reply:
I clicked `Yes`, then got another error - this in console on backend:

```
Request data: {
  conversation_id: 7,
  user_api_key: 'key',
  org_id: 1,
  button_clicked: 'yes'
}
[TypeError: Failed to parse URL from ]
```

Reply:
It succeeds at GET requests still, by the way